core/blockchain.go

```diff
@@ -927,6 +927,23 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 	bc.wg.Add(1)
 	defer bc.wg.Done()
 
+	// Quorum
```

Comment:
We should persist to DB after we acquire the lock `bc.mu.Lock()`

Reply:
changed to persist db after lock has been acquired